### PR TITLE
Correct double click behavior on Mac (maximize)

### PR DESCRIPTION
**Pull request by zbaylin**
- Added `window` to `state` (this might be controversial)
- Added `window` param to `Titlebar`
- Execute the maximize function on double click

### CODE REVIEW

File: src/Model/State.re [hunk 1]
```diff
@@ -61,6 +61,7 @@ type t = {
   terminals: Feature_Terminal.t,
   layout: Feature_Layout.t(int),
   fileExplorer: FileExplorer.t,
+  window: Revery.Window.t,
```
Comment (glennsl) [inline src/Model/State.re:64]: I'm wondering though if we actually want `window` in the state, because it's not serializable. One of the cool things about maintaining all application state in a single data struture is that you can fully re-create an application's state just by serializing it in one place, then move and deserialize it in another. You could for example attach it to a bug report and have the developer re-create the exact conditions of a tricky bug on their own computer. But that doesn't work if the data structure contains unserializable values like functions or opaque external objects like `window` here.

This certainly isn't the only value that is currently unserializable in the state though, so it wouldn't make much difference in itself, but it would be nice if we tried to move toward rather than away from being serializable.

One way we could do this is to maintain the invariant that a workspace has a single window, and keep the mapping from workspace to window outside the state. It could just be a lookup table from workspace path to window instance, for example, so that in order to get the current window of a workspace we'd just call `windowFor(workspacePath)`. Another benefit of this is that it would be a step towards supporting multiple windows. At least as long as we stick with the workspace-window invariant.

Not sure if it's worth the bother just for this though.

File: src/UI/Titlebar.re [hunk 2]
```diff
@@ -44,15 +45,18 @@ module Styles = {
...
+let make =
+    (~isFocused, ~isMaximized, ~title, ~theme, ~window, ~font: UiFont.t, ()) =>
   if (isMaximized) {
     React.empty;
   } else {
     switch (Revery.Environment.os) {
     | Mac =>
-      <View style={Styles.container(~isFocused, ~theme)}>
+      <Clickable
+        onDoubleClick={_ => Revery.Window.maximize(window)}
```
Comment (glennsl) [inline src/UI/Titlebar.re:56]: Actually, I don't think we need to put `window` in the state to implement this either. If the call to `maximize` is moved to a store, the window could just be injected into that store.

The UI shouldn't be causing side-effects anyway, as triggering side-effects anywhere and everywhere tends to make the logic very hard to follow. The UI should only dispatch `action`s (which should be named `event` or `msg` to better convey its actual meaning). The `action` should then be handled by a store (in the old architecture) or `update` function (in the new Feature-based architecture) which will update the `state` and/or trigger side-effects through an `Isolinear.Effect`. This is (approximately) what's usually called [The Elm Architecture](https://guide.elm-lang.org/architecture/).

File: src/Model/State.re (hunk 1 above)
Comment (glennsl) [reply src/Model/State.re:64]: Another benefit of serializable values is that they're also comparable structurally, which is important for some performance optimizations. By checking for structural equality we can avoid re-renders as there's no point in re-rendering if nothing's changes. But if we try comparing values that aren't comparable, we'll crash. Which isn't good.

File: src/UI/Titlebar.re (hunk 2 above)
Comment (glennsl) [reply src/UI/Titlebar.re:56]: Here, for example, you can see `setTitle` being injected into the `TitleStore`:

https://github.com/onivim/oni2/blob/ea560d1d9c4e78d59028f83ca81c7a7d28a54d63/src/Store/TitleStoreConnector.re#L104

You could either pass `window` in as an argument to `start` here as well, or perhaps a `maximize` function to make the store easier to test without creating or faking a window instance.

**Comment by kinglouie** [comment]
I dont know if macOS will do this automatically depending on setting but just to let you know, the double click on titlebar is configurable to either maximize or minimize.

**Comment by bryphe** [comment]
Wow, very impressive turnaround time indeed 🎉 

Looking at CI - it looks like there's a failure building the integration tests with this change: 
```
Generating script.
Script contents:
esy @integrationtest build
========================== Starting Command Output ===========================
/bin/bash --noprofile --norc /home/vsts/work/_temp/1bbf8bd5-6f13-49ff-a5b0-952efebc494f.sh
info esy build 0.6.2 (using integrationtest.json)
    ocamlopt integration_test/lib/.Oni_IntegrationTestLib.objs/native/oni_IntegrationTestLib.{cmx,o} (exit 2)
(cd /home/vsts/work/1/s/_esy/integrationtest/store/b/oni2-1c1e07af/default && /home/vsts/.esy/3_____________________________________________________________________/i/ocaml-4.9.0-031bf83c/bin/ocamlopt.opt -w -40 -g -I integration_test/lib/.Oni_IntegrationTestLib.objs/byte -I integration_test/lib/.Oni_IntegrationTestLib.objs/native -I /home/vsts/.esy/3_____________________________________________________________________/i/brisk__s__brisk_reconciler-5950d842/lib/brisk-reconciler -I /home/vsts/.esy/3_____________________________________________________________________/i/editor_core_types-2d68cb71/lib/editor-core-types -I /home/vsts/.esy/3_____________________________________________________________________/i/editor_input-48d1640e/lib/editor-input -I /home/vsts/.esy/3_____________________________________________________________________/i/flex-1.2.3-4789b89b/lib/flex -I /home/vsts/.esy/3_____________________________________________________________________/i/glennsl__s__timber-1.0.0-54649703/lib/timber -I /home/vsts/.esy/3_____________________________________________________________________/i/isolinear-7a40771c/lib/isolinear -I /home/vsts/.esy/3_____________________________________________________________________/i/ocaml-4.9.0-031bf83c/lib/ocaml/threads -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__astring-opam__c__0.8.3-bec44ec1/lib/astring -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__base-opam__c__v0.13.1-3cc8048e/lib/base -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__base-opam__c__v0.13.1-3cc8048e/lib/base/base_internalhash_types -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__base-opam__c__v0.13.1-3cc8048e/lib/base/caml -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__base-opam__c__v0.13.1-3cc8048e/lib/base/shadow_stdlib -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__biniou-opam__c__1.2.1-a44a33bd/lib/biniou -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__camomile-opam__c__1.0.2-e3ad62da/lib/camomile -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__camomile-opam__c__1.0.2-e3ad62da/lib/camomile/default_config -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__camomile-opam__c__1.0.2-e3ad62da/lib/camomile/dyn -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__camomile-opam__c__1.0.2-e3ad62da/lib/camomile/lib_default -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__camomile-opam__c__1.0.2-e3ad62da/lib/camomile/library -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__charinfo__width-opam__c__1.1.0-f331ed45/lib/charInfo_width -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__ctypes-opam__c__0.15.1-0d50994b/lib/ctypes -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__decoders-opam__c__0.3.0-a99f7683/lib/decoders -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__decoders_yojson-opam__c__0.3.0-ad38e0cc/lib/decoders-yojson -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__easy_format-opam__c__1.3.2-527f1808/lib/easy-format -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__fmt-opam__c__0.8.8-f8d4e4bf/lib/fmt -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__fpath-opam__c__0.7.2-8bd919ea/lib/fpath -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__integers-opam__c__0.4.0-35fb5ef7/lib/integers -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__logs-opam__c__0.7.0-198edb89/lib/logs -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__lru-fdc641fa/lib/lru -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__luv-opam__c__0.5.2-61b28803/lib/luv -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__luv-opam__c__0.5.2-61b28803/lib/luv/c -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__luv-opam__c__0.5.2-61b28803/lib/luv/c_function_descriptions -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__luv-opam__c__0.5.2-61b28803/lib/luv/c_type_descriptions -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__lwt-opam__c__4.5.0-e69a7997/lib/lwt -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__lwt-opam__c__4.5.0-e69a7997/lib/lwt/unix -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__markup-opam__c__0.8.2-fc3937fb/lib/markup -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__mmap-opam__c__1.1.0-5e285863/lib/mmap -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__ocamlfind-opam__c__1.8.1-196676a2/lib/bytes -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__ocplib_endian-opam__c__1.1-72847b1b/lib/ocplib-endian -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__ocplib_endian-opam__c__1.1-72847b1b/lib/ocplib-endian/bigstring -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__ppx__deriving-opam__c__4.4.1-be72bce1/lib/ppx_deriving/runtime -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__ppx__deriving__yojson-opam__c__3.5.2-7e7564d1/lib/ppx_deriving_yojson/runtime -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__psq-opam__c__0.2.0-272afb59/lib/psq -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__re-opam__c__1.9.0-35b21fec/lib/re -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__react-opam__c__1.2.1-a381ebf6/lib/react -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__result-opam__c__1.5-0f7b440f/lib/result -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__seq-opam__c__base-e6ed2c0f/lib/seq -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__sexplib0-opam__c__v0.13.0-1755e674/lib/sexplib0 -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__stdlib_shims-opam__c__0.1.0-9049e772/lib/stdlib-shims -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__uchar-opam__c__0.0.2-34f41375/lib/uchar -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__uucp-opam__c__13.0.0-92d2382d/lib/uucp -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__uutf-opam__c__1.0.2-160f540f/lib/uutf -I /home/vsts/.esy/3_____________________________________________________________________/i/opam__s__yojson-9a770840/lib/yojson -I /home/vsts/.esy/3_____________________________________________________________________/i/reason_font_manager-2.1.1-a928e453/lib/reason-font-manager -I /home/vsts/.esy/3_____________________________________________________________________/i/reason_harfbuzz-1.91.8001-1bd1d359/lib/harfbuzz -I /home/vsts/.esy/3_____________________________________________________________________/i/reason_jsonrpc-1.5.0-6c5694f4/lib/reason-jsonrpc -I /home/vsts/.esy/3_____________________________________________________________________/i/reason_libvim-d12c7f9b/lib/libvim -I /home/vsts/.esy/3_____________________________________________________________________/i/reason_libvterm-4f4f2f32/lib/vterm -I /home/vsts/.esy/3_____________________________________________________________________/i/reason_native__s__console-0.0.3-7190d0a8/lib/console/lib -I /home/vsts/.esy/3_____________________________________________________________________/i/reason_oniguruma-6.94.1002-53ff1fa0/lib/oniguruma -I /home/vsts/.esy/3_____________________________________________________________________/i/reason_sdl2-2.10.3026-9cda14b2/lib/sdl2 -I /home/vsts/.esy/3_____________________________________________________________________/i/reason_skia-9f3acbb2/lib/skia -I /home/vsts/.esy/3_____________________________________________________________________/i/reason_skia-9f3acbb2/lib/skia/wrapped -I /home/vsts/.esy/3_____________________________________________________________________/i/reason_skia-9f3acbb2/lib/skia/wrapped/bindings -I /home/vsts/.esy/3_____________________________________________________________________/i/reason_skia-9f3acbb2/lib/skia/wrapped/c -I /home/vsts/.esy/3_____________________________________________________________________/i/reason_skia-9f3acbb2/lib/skia/wrapped/types -I /home/vsts/.esy/3_____________________________________________________________________/i/reason_textmate-3.1.2-8152a5d9/lib/textmate -I /home/vsts/.esy/3_____________________________________________________________________/i/reason_tree_sitter-1.0.1001-724dcd2c/lib/treesitter -I /home/vsts/.esy/3_____________________________________________________________________/i/reasonfuzz-70499238/lib/Matchers -I /home/vsts/.esy/3_____________________________________________________________________/i/reasonfuzz-70499238/lib/ReasonFuzz -I /home/vsts/.esy/3_____________________________________________________________________/i/rebez-21f8b5cc/lib/rebez/lib -I /home/vsts/.esy/3_____________________________________________________________________/i/rench-d87a41d3/lib/Rench -I /home/vsts/.esy/3_____________________________________________________________________/i/revery-807178f6/lib/Revery -I /home/vsts/.esy/3_____________________________________________________________________/i/revery-807178f6/lib/Revery/Core -I /home/vsts/.esy/3_____________________________________________________________________/i/revery-807178f6/lib/Revery/Draw -I /home/vsts/.esy/3_____________________________________________________________________/i/revery-807178f6/lib/Revery/Font -I /home/vsts/.esy/3_____________________________________________________________________/i/revery-807178f6/lib/Revery/Math -I /home/vsts/.esy/3_____________________________________________________________________/i/revery-807178f6/lib/Revery/Native -I /home/vsts/.esy/3_____________________________________________________________________/i/revery-807178f6/lib/Revery/UI -I /home/vsts/.esy/3_____________________________________________________________________/i/revery-807178f6/lib/Revery/UI_Components -I /home/vsts/.esy/3_____________________________________________________________________/i/revery-807178f6/lib/Revery/UI_Hooks -I /home/vsts/.esy/3_____________________________________________________________________/i/revery-807178f6/lib/Revery/UI_Primitives -I /home/vsts/.esy/3_____________________________________________________________________/i/revery-807178f6/lib/Revery/Utility -I /home/vsts/.esy/3_____________________________________________________________________/i/revery_terminal-e06c3af5/lib/revery-terminal/lib -I /home/vsts/.esy/3_____________________________________________________________________/i/revery_text_wrap-fd8120e1/lib/revery-text-wrap -I src/Components/.Oni_Components.objs/byte -I src/Components/.Oni_Components.objs/native -I src/Core/.Oni_Core.objs/byte -I src/Core/.Oni_Core.objs/native -I src/Core/Kernel/.Kernel.objs/byte -I src/Core/Kernel/.Kernel.objs/native -I src/Core/Utility/.Utility.objs/byte -I src/Core/Utility/.Utility.objs/native -I src/Core/WhenExpr/.WhenExpr.objs/byte -I src/Core/WhenExpr/.WhenExpr.objs/native -I src/Extensions/.Oni_Extensions.objs/byte -I src/Extensions/.Oni_Extensions.objs/native -I src/Exthost/.Exthost.objs/byte -I src/Exthost/.Exthost.objs/native -I src/Exthost/Extension/.Exthost_Extension.objs/byte -I src/Exthost/Extension/.Exthost_Extension.objs/native -I src/Exthost/Protocol/.Exthost_Protocol.objs/byte -I src/Exthost/Protocol/.Exthost_Protocol.objs/native -I src/Exthost/Transport/.Exthost_Transport.objs/byte -I src/Exthost/Transport/.Exthost_Transport.objs/native -I src/Feature/Commands/.Feature_Commands.objs/byte -I src/Feature/Commands/.Feature_Commands.objs/native -I src/Feature/Configuration/.Feature_Configuration.objs/byte -I src/Feature/Configuration/.Feature_Configuration.objs/native -I src/Feature/Editor/.Feature_Editor.objs/byte -I src/Feature/Editor/.Feature_Editor.objs/native -I src/Feature/LanguageSupport/.Feature_LanguageSupport.objs/byte -I src/Feature/LanguageSupport/.Feature_LanguageSupport.objs/native -I src/Feature/Layout/.Feature_Layout.objs/byte -I src/Feature/Layout/.Feature_Layout.objs/native -I src/Feature/Menus/.Feature_Menus.objs/byte -I src/Feature/Menus/.Feature_Menus.objs/native -I src/Feature/Modals/.Feature_Modals.objs/byte -I src/Feature/Modals/.Feature_Modals.objs/native -I src/Feature/Notification/.Feature_Notification.objs/byte -I src/Feature/Notification/.Feature_Notification.objs/native -I src/Feature/SCM/.Feature_SCM.objs/byte -I src/Feature/SCM/.Feature_SCM.objs/native -I src/Feature/Search/.Feature_Search.objs/byte -I src/Feature/Search/.Feature_Search.objs/native -I src/Feature/Syntax/.Feature_Syntax.objs/byte -I src/Feature/Syntax/.Feature_Syntax.objs/native -I src/Feature/Terminal/.Feature_Terminal.objs/byte -I src/Feature/Terminal/.Feature_Terminal.objs/native -I src/Feature/Theme/.Feature_Theme.objs/byte -I src/Feature/Theme/.Feature_Theme.objs/native -I src/Input/.Oni_Input.objs/byte -I src/Input/.Oni_Input.objs/native -I src/Model/.Oni_Model.objs/byte -I src/Model/.Oni_Model.objs/native -I src/Service/FileWatcher/.Service_FileWatcher.objs/byte -I src/Service/FileWatcher/.Service_FileWatcher.objs/native -I src/Service/Font/.Service_Font.objs/byte -I src/Service/Font/.Service_Font.objs/native -I src/Service/Syntax/.Service_Syntax.objs/byte -I src/Service/Syntax/.Service_Syntax.objs/native -I src/Service/Terminal/.Service_Terminal.objs/byte -I src/Service/Terminal/.Service_Terminal.objs/native -I src/Service/Vim/.Service_Vim.objs/byte -I src/Service/Vim/.Service_Vim.objs/native -I src/Store/.Oni_Store.objs/byte -I src/Store/.Oni_Store.objs/native -I src/Syntax/.Oni_Syntax.objs/byte -I src/Syntax/.Oni_Syntax.objs/native -I src/Syntax_Client/.Oni_Syntax_Client.objs/byte -I src/Syntax_Client/.Oni_Syntax_Client.objs/native -I src/Syntax_Server/.Oni_Syntax_Server.objs/byte -I src/Syntax_Server/.Oni_Syntax_Server.objs/native -I src/UI/.Oni_UI.objs/byte -I src/UI/.Oni_UI.objs/native -I vendor/zed/.zed_oni.objs/byte -I vendor/zed/.zed_oni.objs/native -intf-suffix .ml -no-alias-deps -open Oni_IntegrationTestLib__ -o integration_test/lib/.Oni_IntegrationTestLib.objs/native/oni_IntegrationTestLib.cmx -c -impl integration_test/lib/Oni_IntegrationTestLib.re.pp.ml)
File "integration_test/lib/Oni_IntegrationTestLib.re", line 106, characters 19-24:
106 |       <Oni_UI.Root state />,
                         ^^^^^
Error: This expression has type
         window:Revery.Window.t ->
         Oni_IntegrationTestLib__.Types.Model.State.t
       but an expression was expected of type Oni_Model.State.t
```

A few checks that are helpful to run:
- `esy format` <-- format code
- `esy b dune build @check` <-- check for warnings

And then integration tests can be built with:
- `esy '@integrationtest' install`
- `esy '@integrationtest' build`

and run with:
- `esy '@integrationtest' run`

File: src/Model/State.re (hunk 1 above)
Comment (zbaylin) [reply src/Model/State.re:64]: Hey @glennsl! Thanks for the warm welcome! I thought putting the window in the state might be a controversial idea (I said as much in my original PR haha). I am by no means tied to that idea. I will look at the setTitle example you provided and see if I can modify the maximize code to fit that.

Thanks again!

File: src/UI/Titlebar.re (hunk 2 above)
Comment (zbaylin) [reply src/UI/Titlebar.re:56]: Gotcha, thank you so much for the explanation. Reworking my code now!

**Comment by zbaylin** [comment]
@kinglouie where is this configuration option specified? That's very interesting -- I never knew that.

File: src/UI/Titlebar.re (hunk 2 above)
Comment (zbaylin) [reply src/UI/Titlebar.re]: I think I managed to figure it out in my latest commit. Let me know if there is still anything that needs to be changed!

File: src/Model/Actions.re [hunk 3]
```diff
@@ -187,6 +187,7 @@ type t =
   | Modals(Feature_Modals.msg)
   // "Internal" effect action, see TitleStoreConnector
   | SetTitle(string)
+  | Maximize
```
Comment (glennsl) [inline src/Model/Actions.re:190]: This should be named like an event, not like a function. The reason is that the decision of what to do should be taken in the store, not in the View. For example, how would you now implement the macOS configuration setting that minimizes on double click instead?

The View should only dispatch messages that describe what happened. The store/update function is responsible for figuring out what to actually do based on that.

I realize there are plenty of examples to the contrary here. That's old debt, and I think part of the reason they are treated as function calls, or "actions", is that the type is actually called `action`. I think it would be easier to understand if it was named `event` or `msg`. And if we got rid of all the other function-call-actions, of course.

My suggestion:
```suggestion
  | TitleDoubleClicked
```

File: src/Store/TitleStoreConnector.re [hunk 4]
```diff
@@ -113,6 +113,11 @@ let start = setTitle => {
       }
     );
 
+  let internalMaximizeEffect = () =>
+    Isolinear.Effect.createWithDispatch(~name="maximize", _dispatch =>
+      maximize()
+    );
```
Comment (glennsl) [inline src/Store/TitleStoreConnector.re:119]: Minor nit. This doesn't need to be a function. You can also use just `create` instead of `createWithDispatch` 

```suggestion
  let internalMaximizeEffect =
    Isolinear.Effect.create(~name="maximize", () =>
      maximize()
    );
```

File: src/Model/Actions.re (hunk 3 above)
Comment (zbaylin) [reply src/Model/Actions.re:190]: Gotcha that makes perfect sense. Thanks again. Will push a fix in a sec.

File: src/Store/TitleStoreConnector.re (hunk 4 above)
Comment (zbaylin) [reply src/Store/TitleStoreConnector.re]: I fixed this in the latest commit. I'll have to browse the Isolinear source sometime today to understand it better. Thanks!

File: src/Model/Actions.re (hunk 3 above)
Comment (zbaylin) [reply src/Model/Actions.re]: Renamed in the latest commit!

File: src/Store/TitleStoreConnector.re [hunk 5]
```diff
@@ -113,10 +113,8 @@ let start = (setTitle, maximize) => {
       }
     );
 
-  let internalMaximizeEffect = () =>
-    Isolinear.Effect.createWithDispatch(~name="maximize", _dispatch =>
-      maximize()
-    );
+  let internalMaximizeEffect =
+    Isolinear.Effect.create(~name="maximize", _dispatch => maximize());
```
Comment (glennsl) [inline src/Store/TitleStoreConnector.re:117]: One more tiny thing:

```suggestion
    Isolinear.Effect.create(~name="maximize", () => maximize());
```

Comment (zbaylin) [reply src/Store/TitleStoreConnector.re:117]: oops! sorry about that haha

Comment (zbaylin) [reply src/Store/TitleStoreConnector.re]: Fixed now!

**Comment by zbaylin** [comment]
I just found the config flag that @kinglouie was referring to. It's under System Preferences -> Dock -> Double click a window's title bar to [zoom | minimize]. I dont know if this information is exposed to the user/available anywhere. I can look around and see though!

**Comment by zbaylin** [comment]
Alright I found that the output of `defaults read "Apple Global Domain" AppleActionOnDoubleClick` is either `None`, `Maximize`, or `Minimize`. We could read the output on launch and determine the behavior then, or we could check it every resize.

Thoughts? @glennsl

**Comment by bryphe** [comment]
> . We could read the output on launch and determine the behavior then, or we could check it every double click.

Cool! My preference would be to query on double-click of the title-bar, just to minimize the work we do on start-up.

**Comment by zbaylin** [comment]
Works for me! I can wire that up right now.

**Comment by zbaylin** [comment]
Unfortunately it looks like a minimize function hasn't been added to revery even though a maximize one has. I can make a PR for that, but maybe it makes sense to merge in as-is and try to fix the behavior as a function of the configuration later?

**Comment by bryphe** [comment]
> I can make a PR for that, but maybe it makes sense to merge in as-is and try to fix the behavior as a function of the configuration later?

Sounds reasonable to me - go ahead and merge 👍
